### PR TITLE
Support sort_key for vpc_offerings table

### PR DESCRIPTION
**Pull request by anuragaw**
Fixes- #2742

UI Supported ordering VPC Offerings but the API did not have that
support implemented. This makes the change in updateVPCOfferings
and listVPCOfferings API calls, along with necessary database
changes for supporting sorting of VPC Offerings.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

## Screenshots (if appropriate):
Added column to Database -
Screenshot from 2019-04-08 15-18-35
## How Has This Been Tested?

Load the UI and sort VPC Offerings. Ensure the order persists. Make sure issue 2742 is not reproducible.

### CODE REVIEW

**Comment by anuragaw** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql [hunk 1]
```diff
@@ -19,3 +19,16 @@
...
+		IN in_table_name VARCHAR(200)
+    , IN in_column_name VARCHAR(200)
+    , IN in_column_definition VARCHAR(1000)
+)
+BEGIN
+
+    DECLARE CONTINUE HANDLER FOR 1091 BEGIN END; SET @ddl = CONCAT('DROP INDEX ', in_index_name); SET @ddl = CONCAT(@ddl, ' ', ' ON ') ; SET @ddl = CONCAT(@ddl, ' ', in_table_name); PREPARE stmt FROM @ddl; EXECUTE stmt; DEALLOCATE PREPARE stmt; END;
+
+-- Add For VPC flag
+CALL `cloud`.`IDEMPOTENT_ADD_COLUMN`('cloud.vpc_offerings','sort_key', 'INT(32) NOT NULL DEFAULT 0');
```
Comment (rhtyd) [inline engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: We can simply add the column, we don't expect past versions to have the column.

**Comment by rhtyd** [comment]
Minor comments raised, lgtm otherwise.

File: engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql (hunk 1 above)
Comment (anuragaw) [reply engine/schema/src/main/resources/META-INF/db/schema-41200to41300.sql]: Fixed in the latest commit @rhtyd

**Comment by anuragaw** [comment]
@resmo - please take a look.

**Comment by anuragaw** [comment]
@rafaelweingartner - a quick look would be very helpful. Thanks! Much appreciated.

File: api/src/main/java/com/cloud/network/vpc/VpcProvisioningService.java [hunk 2]
```diff
@@ -41,12 +43,9 @@ public VpcOffering createVpcOffering(String name, String displayText, List<Strin
...
     /**
-     * @param vpcOffId
-     * @param vpcOfferingName
-     * @param displayText
-     * @param state
+     * @param vpcOfferingCmd
      * @return
      */
-    public VpcOffering updateVpcOffering(long vpcOffId, String vpcOfferingName, String displayText, String state);
+    public VpcOffering updateVpcOffering(final UpdateVPCOfferingCmd vpcOfferingCmd);
```
Comment (DaanHoogland) [inline api/src/main/java/com/cloud/network/vpc/VpcProvisioningService.java]: not sure if it is a biggy but,... any external plugins may suffer from this service api. better to add a new method and mark it deprecated

Comment (anuragaw) [reply api/src/main/java/com/cloud/network/vpc/VpcProvisioningService.java]: Pretty sure there wasn't any internal plugin Daan, but I think it's a good practice in general. Doing the change now.

Comment (anuragaw) [reply api/src/main/java/com/cloud/network/vpc/VpcProvisioningService.java]: Fixed @DaanHoogland . Please take a look.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@anuragaw fixed conflict, fyi

**Comment by anuragaw** [comment]
Thank you @rhtyd !

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Pull request merged**